### PR TITLE
Update for current varnish-cache and polish

### PR DESCRIPTION
**Pull request by nigoroll**
This makes the vmod work and pass the test-suite with varnish-cache as of baf60b4d2ef5f418e6ac470f3562e43c37e2ad44

Please see individual  commit messages.

### CODE REVIEW

File: src/tests/test11.vtc [hunk 1]
```diff
@@ -15,8 +15,8 @@ varnish v1 -vcl+backend {
 			curl.set_timeout(200);
 			curl.set_connect_timeout(200);
 		} else {
-			curl.set_timeout(1300);
-			curl.set_connect_timeout(1300);
+			curl.set_timeout(1310);
```
Comment (gquintard) [inline src/tests/test11.vtc:18]: I have a hard time believing that 10ms make a clean difference in general. Is there a known reason for that?

**Comment by gquintard** [comment]
looks good to me, but I don't have commit access anymore :-)

File: src/tests/test11.vtc (hunk 1 above)
Comment (nigoroll) [reply src/tests/test11.vtc:18]: The reason why I changed the timeout was that the test case reported a run time of 1.299something seconds. This seems a bit odd, but I did not consider it too relevant, have not looked at what libcurl does and just applied this change to make the test case pass.

**Comment by gquintard** [comment]
@Dridi, @hermunn, unless there's an objection, I'm merging this next week

**Comment by dridi** [comment]
I object to @gquintard's definition of a week.

**Comment by gquintard** [comment]
merged, thanks @nigoroll !

**Pull request merged**